src/core/entity/event.hpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023, KNS Group LLC (YADRO)
+
+#pragma once
+
+#include <core/entity/entity_interface.hpp>
+
+#include <functional>
+#include <vector>
+
+namespace app
+{
+
+template <typename TEnumEvent>
+class Event : public virtual IEvent<TEnumEvent>
+{
+    using Subscriber = typename IEvent<TEnumEvent>::Subscriber;
+
+  public:
+    ~Event() override = default;
+
+    /**
+     * @brief Add subscriber to observe query process events.
+     *
+     * @note thread unsafe
```

Comment:
As this is a public method shouldn't we add a mutex here?

Reply:
We don't known how this both methods will be used (yes I know has it will:) and this is because I don't add a mutex and mark their as thread-safe) and this undefineing may follow to self-blocking in some cases (the Subscribe kind is lambda).